kanidmd/idm/src/actors/v1_write.rs

```diff
@@ -778,7 +778,7 @@ impl QueryServerWriteV1 {
             let intent_token = CredentialUpdateIntentToken {
                 intent_id: intent_token.token,
             };
-
+            // TODO: this is throwing a 500 error when a session is already in use, that seems bad?
```

Comment:
It's the intended process because we don't want the "intent token" to be used multiple times, so if it's already "been exchanged" we have to wait for that previous session to be completed, timeout, or commited, to know if we can proceed. What situation are you seeeing the 500 under?

Reply:
I issued the token, started the `use_reset_token` flow - but ctrl-c'd out of the REPL, then ran it again straight away. The server logged "in use" in the console - the note's to remind me to dig into why it threw a 500. 😄

Reply:
We could persist the session in this case an detect the resumption?

Reply:
I guess so, fun edge case either way

Reply:
I think it's valid as an issue. It can certainly be improved. I'll open it now.